content/tutorials/peer-to-peer-vue.textile

```diff
@@ -0,0 +1,671 @@
+
+After 2 minutes have elapsed, recovery is no longer possible and the connection will move to the suspended state.
+
+The suspended state is entered after a failed connection attempt if there has then been no connection for a period of two minutes. In the suspended state, the library will periodically attempt to open a new connection every 30 seconds. Developers are unable to publish messages in this state. A new connection attempt can also be triggered by an explicit call to @connect()@ on the Connection object.
+
+Once the connection has been re-established, channels will be automatically re-attached. The client has been disconnected for too long for them to resume from where they left off, so if it wants to catch up on messages published by other clients while it was disconnected, it needs to use the history API.
+
+p. So much like our @P2PServer@ will buffer outbound messages, any disconnected clients will receive all the messages sent in a window of up to **two minutes of disconnection**
+
+h3. Using the History API to catch up
+
+Ably offers a @History API@ and a @Rewind@ setting. The History API allows you to query the history of a channel for by default, the last *two minutes* of data. If you enable @persisted history@ on your channel, that window is extended for 24-72 hours (these will count towards your message quotas - "see docs for more on this":https://support.ably.com/support/solutions/articles/3000030059). We can use this extended window to allow clients to join "mid-stream" and catch up when they connect.
+
+The "Rewind":https://www.ably.io/documentation/realtime/channels/channel-parameters/overview#supported-parameters parameter can be enabled when you subscribe to a channel, which will immediatey then receive up to the last 100 messages in the time window you specify.
+
+The caveat is that the client will be processing historical messages - so be sure you understand what that means for your specific application. In some scenarios, it may be more effective for another @peer@ or the @server@ to just send new clients the state they **need** rather than the client reconstructing it.
+
+To enable Rewind, we need to add an additional parameter to the @channels.get@ call, to add the parameter @ { params: { rewind: '1m' } }`. This tells our client to return up to the last 100 messages, from the minute. You can provide different time periods in here in either minutes ('2m') or seconds ('15s').
+
+```[javascript]
+async connect(identity, uniqueId) {
+  ...
+  const ably = new Ably.Realtime.Promise({ authUrl: '/api/createTokenRequest' });
+  this.channel = await ably.channels.get(`p2p-sample-${uniqueId}`, { params: { rewind: '1m' } });  
+  ...
+}
+```
+
+You will now have a working peer to peer base on which you can build all kinds of apps and games!
+
+h3. But wait, there's more!
+
+This is a self contained demo of building peer to peer apps, hosting in browser tabs, using @Ably Channels@ as their communication medium.
```

Comment:
May be me mis-reading, but I think `building peer to peer apps, hosting in browser tabs` => `building peer to peer apps, hosted in browser tabs`

Reply:
yeah, its not super clear is it, I meant that its a demo of hosting a session in a browser tab